### DISCUSSION TITLE
on pull_request labeled

### DISCUSSION THREAD

**Discussion by dudo** [community/community#25819] [Actions]
<p>We can tell an action to run when a label is added (or removed), but I can’t seem to narrow the action to _which_ label was added.</p>
<p>Anyone know how to accomplish this? With the hcl syntax, GitHub recommended using the filter action to stop a run. The yaml syntax is way better, but this functionality seems to be missing.</p>

**Comment by BrightRan** [comment, marked as answer]
<p>Hi <a class="mention" href="/dudo">@dudo</a>,</p>
<p>At this moment, it does not support to set filter of labels on the label event. Any label is created, edited or deleted will trigger the workflow to run.</p>
<p>As a workaround, you syntax  <strong>jobs.&lt;job_id&gt;.if</strong>  to set conditions to decide which labels can let the jobs run or not.</p>
<p>For example:</p>
<p>The labels _ <strong>testLab01</strong> _ and _ <strong>testLab02</strong> _ are created, edited or deleted will not let  <strong>first_job</strong> be executed.</p>
<pre><code>jobs:
first_job:
if: github.event_name == 'label' &amp;&amp; github.event.label.name != 'testLab01' &amp;&amp; github.event.label.name != 'testLab02'
name: First Job
</code></pre>
<p>Only the label _ <strong>bug</strong> _ is created, edited or deleted will let <strong>second_job</strong> be executed.</p>
<pre><code>jobs:
  second_job:
    if: github.event_name == 'label' &amp;&amp; github.event.label.name == 'bug'
    name: Second Job
</code></pre>
<p><strong>Note:</strong></p>
<ol>
<li>
<p>This workaround can’t prevent the workflow from running, the workflow will still run. It just can prevent the jobs from running during runing the workflow.</p>
</li>
<li>
<p>If you want the conditions can be available to all jobs, you need to add the conditions on every job.</p>
</li>
</ol>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/270ff8c221b0b7c810e8da5206e05ca6d07be857.png" data-download-href="/uploads/short-url/5zyQ7cuvgEi2tYEX8sUpQOcYpfN.png?dl=1" title="example.PNG" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/46307996/181093610-ebb18568-1903-48e7-a501-874de4e8e982.png" alt="example.PNG" data-base62-sha1="5zyQ7cuvgEi2tYEX8sUpQOcYpfN" width="690" height="396" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/270ff8c221b0b7c810e8da5206e05ca6d07be857_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">example.PNG</span><span class="informations">861×495 9.34 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by dudo** [comment]
<p>Seems like you might be on the team - any plans on adding this functionality? Your solution is very similar to the original “filter” solution, but more robust label support would be pretty powerful.</p>
